gwt-ol3-client/src/main/java/ol/OLUtil.java

```diff
@@ -177,11 +177,15 @@ public void removeHandler() {
      * @return {@link HandlerRegistration}
      */
     public static HandlerRegistration addMapZoomListener(final Map map, final MapZoomListener listener) {
-        return observe(map.getView(), "propertychange", new EventListener<ObjectEvent>() {
+        return observe(map, "moveend", new EventListener<ObjectEvent>() {
 
+	    private int zoomLevel = (int) map.getView().getZoom();
+		
             @Override
             public void onEvent(ObjectEvent event) {
-                if("resolution".equals(event.getKey())) {
+                int newZoomLevel = (int) map.getView().getZoom();
```

Comment:
Casting to int isn't sufficiant if you are using fractional zoom. Therefore casting to double is nesseccary.